### DISCUSSION TITLE
Improvement in API credit use

### DISCUSSION THREAD

**Discussion by ziogref** [ThioJoe/YT-Spammer-Purge#821] [Ideas]
Looking at the wiki, it says that 

> Each set of 5 recent videos listed costs 100 units (before any comment scanning is even done)

I have found a way to reduce that to 1 api unit. 

ThioJoe's Channel ID is `UCQSpnDG3YsFNf5-qHocF-WQ`

If you change the 2nd character from a C to a U you get the ID of the hidden `uploads` playlist
Using Youtube's API for retrieving playlists does not show this playlists existence.

So in the case of ThioJoe it would be `UUQSpnDG3YsFNf5-qHocF-WQ`

So using an API call such as
`https://youtube.googleapis.com/youtube/v3/playlistItems?part=snippet%2CcontentDetails&maxResults=5&playlistId=[Playlist ID]&key=[API Key]`

you can retrieve the 5 latest videos (or more) only using 1 api credit

This also retrieves shorts.

**Comment by DinhHuy2010** [comment]
that is on 2.17 branch and look at https://github.com/ThioJoe/YT-Spammer-Purge/pull/786

**Comment by ziogref** [comment]
ah cool, I wasn't sure as the wiki still said 100 credits.
